web/libs/ui/src/lib/empty-state/empty-state.tsx

```diff
@@ -0,0 +1,322 @@
+    const iconSize = iconSizes[size];
+    const iconWithSize = React.cloneElement(icon as React.ReactElement, {
+      width: iconSize.width,
+      height: iconSize.height,
+    });
+
+    // Get typography configuration for the size
+    const titleConfig = titleVariants[size];
+    const descriptionConfig = descriptionVariants[size];
+
+    // Generate unique IDs for accessibility
+    const titleElementId = `empty-state-title-${Math.random().toString(36).substr(2, 9)}`;
+    const descriptionElementId = `empty-state-desc-${Math.random().toString(36).substr(2, 9)}`;
+
+    return (
+      <div
+        ref={ref}
+        className={cn("empty-state", styles.base, sizes[size], variants[variant], className)}
+        data-testid={testId}
+        aria-label={ariaLabel}
+        aria-labelledby={!ariaLabel ? titleElementId : undefined}
+        aria-describedby={descriptionElementId}
+        {...rest}
+      >
+        {/* Icon */}
+        <div className={cn("empty-state__icon", styles.icon, "flex items-center justify-center rounded-full")}>
+          {iconWithSize}
+        </div>
+
+        {/* Title and Description - Special layout for small variant */}
+        {size === "small" ? (
+          <div className={cn("empty-state__text-wrapper", styles["text-wrapper"])}>
+            <Typography
+              {...titleConfig}
+              className={cn("empty-state__title", styles.title, "text-neutral-content")}
+              id={titleElementId}
+            >
+              {title}
+            </Typography>
+
+            <Typography
+              {...descriptionConfig}
+              className={cn("empty-state__description", styles.description, "text-neutral-content-subtler")}
+              id={descriptionElementId}
+            >
+              {description}
+            </Typography>
+          </div>
+        ) : (
+          <>
+            {/* Standard layout for large and medium */}
+            <Typography {...titleConfig} className={cn("empty-state__title", styles.title)} id={titleElementId}>
+              {title}
+            </Typography>
+
+            <Typography
+              {...descriptionConfig}
+              className={cn("empty-state__description", styles.description, "text-neutral-content-subtler")}
+              id={descriptionElementId}
+            >
+              {description}
+            </Typography>
+          </>
+        )}
+
+        {/* Additional Content */}
+        {additionalContent && (
+          <div className={cn("empty-state__additional", styles.additional)}>{additionalContent}</div>
+        )}
+
+        {/* Actions */}
+        {actions && (
+          <div className={cn("empty-state__actions", styles.actions)}>
+            {(() => {
+              // Flatten children and filter out null/false values to count actual rendered elements
+              const flattenedActions = React.Children.toArray(actions).flat().filter(Boolean);
```

Comment:
it's already a flat array

```suggestion
              // Filter out null/false values to count actual rendered elements
              const realActions = React.Children.toArray(actions).filter(Boolean);
```